linkerd/tls/src/server/mod.rs

```diff
@@ -196,20 +200,27 @@ where
     let mut buf = [0u8; PEEK_CAPACITY];
     let sz = io.peek(&mut buf).await?;
     debug!(sz, "Peeked bytes from TCP stream");
-    match conditional_accept::match_client_hello(&buf, &local_id) {
-        conditional_accept::Match::Matched => {
+    match client_hello::parse_sni(&buf) {
+        Ok(Some(ServerId(sni))) if sni == local_id => {
             trace!("Identified matching SNI via peek");
             // Terminate the TLS stream.
-            let (peer_id, tls) = handshake(tls_config, PrefixedIo::from(io)).await?;
-            return Ok((peer_id, EitherIo::Right(tls)));
+            let (client_id, io) = handshake(tls_config, PrefixedIo::from(io)).await?;
+            let tls = Conditional::Some(Tls::Terminated { client_id });
+            return Ok((tls, EitherIo::Right(io)));
         }
 
-        conditional_accept::Match::NotMatched => {
+        Ok(Some(sni)) => {
+            trace!("Identified non-matching SNI via peek");
```

Comment:
nit: should this include the SNI? I guess the `parse_sni` function also logs it, but this module's logs might be enabled while `client_hello`'s might not be...